src/generic/Vector.h

```diff
@@ -150,6 +150,17 @@ namespace oomph
     {
     }
 
+    /// \short A constructor that creates a vector of variable size (using an
+    /// initialiser_list) with entries set to the values in the initialiser_list
```

Comment:
I would say that `of variable size (using an initialiser_list)` is redundant and potentially confusing.

Also I seen you've added the ability to set the allocator, but a comment above says `The generalisation to general allocators is NOT handled here`. Might be worth checking with @MatthiasHeilManchester or @alhazel about whether this should still be the case.

Reply:
Good point, I've stripped it from the comment. Thanks!

I hadn't noticed that - that's interesting one. I _did_ make an effort to avoid adding `std::allocator<T>` arguments everywhere else (as done with `std::vector`), so it will be the only constructor that takes an `std::allocator<T>`. Also, they may not want to generalise it to all possible STL constructors (e.g. move constructors), but I think this one is far too helpful to pass up!

Reply:
??? Somebody will have to explain this to me!

Reply:
**Allocators**

The `std::vector` class typically takes two values: (1) the element type, and (2) the allocator type (which has a default; `std::allocator<T>`). The `Vector` wrapper around `std::vector` drops the second template argument and only specifies the element type. Any calls to the underlying `std::vector` class therefore only need to specify the element type and the defaults specified by `std::vector` handle the allocator selection. I _think_ this is what the comment about "not generalising to general allocators" refers to.

After writing this out, I realise that I can actually omit the `std::allocator<_Tp>` argument, making the code even simpler, so I've changed that, and recommend waiting for the self-tests to pass.

**Initialiser lists**

An `initializer_list<T>` is a container introduced in C++11 that can be created using curly braces `{}` and can be passed any number of entries (of the same type!) and it will create a container of the appropriate size. Many (or possibly all; I'm not sure) STL containers have a constructor that allows a user to initialise using an `initializer_list<T>`. It also lets users use a new type of initialisation called "list initialisation" which does not allow narrowing conversions (which can be quite nice), e.g.

```cpp
// Okay
int a{1};

// Error!
// temp.cc:3:9: error: type 'double' cannot be narrowed to 'int' in initializer list [-Wc++11-narrowing]
//  int a{1.0};
int a{1.0};
```
More details
- [modern-cpp-features](https://github.com/AnthonyCalandra/modern-cpp-features#initializer-lists) 
- [list initialisation](https://en.cppreference.com/w/cpp/language/list_initialization).

Reply:
> Good point, I've stripped it from the comment. Thanks!

Did you? Or rather did you do it and then push your change? Can't see it!

Reply:
...not sure what you're talking about, it's all definitely changed 👀